build/Build.md

```diff
@@ -0,0 +1,190 @@
+
+Useful reference points:
+
+[1] [Docker resources for WSO2 API Management](https://github.com/wso2/docker-apim) <br/>
+[2] [Docker resources for WSO2 Integration](https://github.com/wso2/docker-ei) <br/>
+[3] [Docker resources for WSO2 Identity and Access Management](https://github.com/wso2/docker-is)
+
+<br/>
+
+## 1.4. Tips to harden your image
+
+Key takeaways from our experience to write your dockerfile :
+
+```
+[1] Verify Authenticity of any software installed in the image
+[2] Always use HTTPS for files added remotely
+[3] From directive: Use specific / unique tag
+[4] Never run as root
+[5] USER directive: Always
+[6] Drop privileges where necessary
+[7] Expose only required ports
+```
+
+Useful reference points:
+
+[1] [Docker resources for WSO2 API Management](https://github.com/wso2/docker-apim) <br/>
+[2] [Docker resources for WSO2 Integration](https://github.com/wso2/docker-ei) <br/>
+[3] [Docker resources for WSO2 Identity and Access Management](https://github.com/wso2/docker-is)
+
+<br/>
+
+## 1.5. Tips on Tagging
+
+WSO2 product docker image tagging strategy is simple.
+
+[1] Per each product version, we maintain an image queue. <br/>
+This queue can hold a maximum of 3 images from the oldest to the latest and each bear a unique tag to distinguish itself from the other.
+
+* *E.g.* 	
+        <br/>
+        *wso2am:3.1.0.1591023165861.2 - Oldest* <br/>
+        *wso2am:3.1.0.1591230650206.3 - 2nd oldest* <br/>
+	    *wso2am:3.1.0.1592240780334.4 - latest* <br/>
+
+        Unique tag: <product>:<version>.<wso2-update-timestamp>.<docker-tag-last-digit>
+
+This queue is regularly updated with a new image having an update (or in other words, a patch) either to the product or docker artifacts. When a new image is pushed, the oldest of the 3 is removed from the registry on a first-in-first-out (FIFO) basis. 
+
+Unique tags are ideal in referring to an image during deployments and we recommend customers to use this tag when they directly use WSO2 images for their deployments without any customizations.
+
+[2] We also maintain one stable tag per each product version. <br/>
+A stable tag is always pointed to the latest docker image of a particular product version. 
+
+Taking the same example from [1], 
+
+* *E.g.* 	
+        <br/>
+        *wso2am:3.1.0.1591023165861.2 - Oldest* <br/>
+        *wso2am:3.1.0.1591230650206.3 - 2nd oldest* <br/>
+	    *wso2am:3.1.0.1592240780334.4 (unique), wso2am:3.1.0 (Stable) - latest* <br/>
+
+        Stable tag: <product>:<version>
+
+Stable tags are ideal in referring to a base image during docker build. We recommend customers to use this tag when they want to build their custom images on top of the product docker images that we host.
+
+[3] We maintain a latest tag per each product. <br/>
+The latest tag is also a stable tag and is always pointed to the latest docker image of a product repo.
+
+Taking the same example from [2], if 3.1.0 is the latest version of wso2am,
```

Comment:
@DilanUA shall we highlight the versions with back-ticks (e.g. `3.1.0`)?

Reply:
let's do this as an improvement in a latter PR